### ISSUE TITLE
Deleting multiple folders too quickly deletes literally everything

### ISSUE THREAD

**Issue by W2Wizard** [FlaxEngine/FlaxEngine#210]
**Issue description:**
When attempting to delete multiple folders too quickly, it can result in deleting literally the entire project.

**Steps to reproduce:**
1. Have lots of folders present in your project:
 ![image](https://user-images.githubusercontent.com/63303990/106958013-118a3880-6739-11eb-97cf-f2043e72bb89.png)
2. Now attempt to select and delete them as quick as you can
3. Get dialog asking to delete  ' ' , click Ok.
![image](https://user-images.githubusercontent.com/63303990/106958212-57df9780-6739-11eb-85cb-46aed155c6e2.png)
4. Now its all gone.
![unknown](https://user-images.githubusercontent.com/63303990/106957824-cec86080-6738-11eb-9d56-3fb1605644cd.png)

**Flax version:**
1.0.6216

**Comment by SilentCLD** [comment]
So I did a bit of research into this, it seems to be an issue with the folder path being an empty string after a folder is deleted and before another is selected.
You don't have to delete folders quickly, in fact you can delete a folder and as long as you don't select a new one when you press the Del key the issue will occur.

To anyone attempting to fix the issue:
https://github.com/FlaxEngine/FlaxEngine/blob/master/Source/Editor/Windows/ContentWindow.cs#L406
A quick fix would be to check the path in ContentWindow.cs and remove the item from the toDelete list if it's empty, something like the following:
```
for (int i = 0; i < toDelete.Count;)
{
    var item = toDelete[i];

    // Path can be empty if the user deleted an item and pressed the delete key without selecting a new one
    if (string.IsNullOrWhiteSpace(item.Path))
    {
        Editor.LogWarning("Tried to remove folder with no path.");
        toDelete.RemoveAt(i);
        continue;
    }

    i++;
}

// If all items were invalid just return.
if (toDelete.Count == 0)
    return;
```

A more robust solution would be to only allow it if the path is part of the project directory, this would also stop the problem of being able to delete the engine install directory, this isn't really a problem for official installs as they're in a protected dir but more for source builds (I haven't tried it but it gives me the option😄)

**Comment by mafiesto4** [comment]
Fixed by #212
